examples/stm32/f1/stm32-h103/button/button.c

```diff
@@ -62,8 +60,7 @@ int main(void)
 	while (1) {
 		gpio_toggle(GPIOC, GPIO12);
 
-		exti_line_state = GPIOA_IDR;
-		if ((exti_line_state & (1 << 0)) != 0) {
+		if ((gpio_get(GPIOA, GPIO0) & GPIO0) == 1) {
```

Comment:
This is redundant.  gpio_get is implemented as gpio_port_read(port) & pin.  Simply
```C
if (gpio_get(GPIOA, GPIO0)) {
```
is sufficient.  The == 1 is misleading as well, it only works for gpio0.